File: frontend/src/interfaces/Comment.ts
// src/interfaces/Comment.ts
import type { User } from "./User";
import type { Thread } from "./Thread";

export interface Comment {
  ID: number;
  content: string;
  user_id: number;
  user?: User;
  thread_id: number;
  thread?: Thread;
  parent_comment_id?: number | null;
  parent?: Comment;
  replies?: Comment[];
}

export interface CreateCommentRequest {
  content: string;
  user_id: number;
  thread_id: number;
  parent_comment_id?: number | null;
}

export interface UpdateCommentRequest {
  ID: number;
  content?: string;
  parent_comment_id?: number | null;
}

File: backend/entity/comment.go
package entity

import (
	"gorm.io/gorm"
)

type Comment struct {
	gorm.Model
	Content          string    `json:"content"`
	UserID           uint      `json:"user_id"`
	User             *User     `gorm:"foreignKey:UserID" json:"user"`
	ThreadID         uint      `json:"thread_id"`
	Thread           *Thread   `gorm:"foreignKey:ThreadID" json:"thread"`
	ParentCommentID  *uint     `json:"parent_comment_id"` // เป็น nil ถ้าเป็นคอมเมนต์ระดับบน
	Parent           *Comment  `gorm:"foreignKey:ParentCommentID" json:"parent"`

	Replies []Comment `gorm:"foreignKey:ParentCommentID" json:"replies,omitempty"`
}